src/formatDate.js

```diff
@@ -8,7 +8,76 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArray = [];
+  let day = '';
+  let month = '';
+  let year = '';
+
+  if (date.includes('-')) {
+    dateArray = date.split('-');
+  }
+
+  if (date.includes('/')) {
+    dateArray = date.split('/');
+  }
+
+  if (fromFormat[0] === 'YYYY' || fromFormat[0] === 'YY') {
+    year = dateArray[0];
+    month = dateArray[1];
+    day = dateArray[2];
+  } else if (fromFormat[1] === 'YYYY') {
+    year = dateArray[1];
+    month = dateArray[0];
+    day = dateArray[2];
+  } else {
+    year = dateArray[2];
+    month = dateArray[1];
+    day = dateArray[0];
+  }
+
+  if (toFormat[0] === 'YYYY' && year.length === 4) {
+    return year + toFormat[3] + month + toFormat[3] + day;
+  }
+
+  if (toFormat[0] === 'YY' && year.length === 4) {
+    return year.slice(2) + toFormat[3] + month + toFormat[3] + day;
+  }
+
+  if (toFormat[0] === 'YY' && year.length === 2) {
+    return year + toFormat[3] + month + toFormat[3] + day;
```

Comment:
Again, the assumption about the separator at index 3 of the `toFormat` array may not be correct. Adjust the logic to handle different formats appropriately.